includes/settings-page.php

```diff
@@ -13,94 +13,96 @@
+			'callback' => 'rt_scripts_optimizer_style_dequeue_non_logged_handles_callback',
+			'page'     => 'rt-scripts-optimizer-settings',
+			'section'  => 'rt_scripts_optimizer_settings_section'
+		],
+		'rt_scripts_optimizer_style_async_handles'              => [
+			'id'       => 'rt_scripts_optimizer_style_async_handles',
+			'title'    => __( 'CSS handles of the stylesheets which should be asynchronously loaded', 'RT_Script_Optimizer' ),
+			'callback' => 'rt_scripts_optimizer_style_async_handles_callback',
+			'page'     => 'rt-scripts-optimizer-settings',
+			'section'  => 'rt_scripts_optimizer_settings_section'
+		],
+		'rt_scripts_optimizer_style_async_handles_onevent'      => [
+			'id'       => 'rt_scripts_optimizer_style_async_handles_onevent',
+			'title'    => __( 'CSS handles of the stylesheets which should be asynchronously loaded on any window event', 'RT_Script_Optimizer' ),
+			'callback' => 'rt_scripts_optimizer_style_async_handles_onevent_callback',
+			'page'     => 'rt-scripts-optimizer-settings',
+			'section'  => 'rt_scripts_optimizer_settings_section'
+		],
+		'rt_scripts_optimizer_load_amp_boilerplate_style'       => [
+			'id'       => 'rt_scripts_optimizer_load_amp_boilerplate_style',
+			'title'    => __( 'Load AMP boilerplate CSS', 'RT_Script_Optimizer' ),
+			'callback' => 'rt_scripts_optimizer_load_amp_boilerplate_style_callback',
+			'page'     => 'rt-scripts-optimizer-settings',
+			'section'  => 'rt_scripts_optimizer_settings_section'
+		],
+		'rt_scripts_optimizer_skip_css_concatination_all'       => [
+			'id'       => 'rt_scripts_optimizer_skip_css_concatination_all',
+			'title'    => __( 'Skip all CSS concatination', 'RT_Script_Optimizer' ),
+			'callback' => 'rt_scripts_optimizer_skip_css_concatination_all_callback',
+			'page'     => 'rt-scripts-optimizer-settings',
+			'section'  => 'rt_scripts_optimizer_settings_section'
+		],
+		'rt_scripts_optimizer_skip_css_concatination_handles'   => [
+			'id'       => 'rt_scripts_optimizer_skip_css_concatination_handles',
+			'title'    => __( 'Skip CSS concatination for these handles', 'RT_Script_Optimizer' ),
+			'callback' => 'rt_scripts_optimizer_skip_css_concatination_handles_callback',
+			'page'     => 'rt-scripts-optimizer-settings',
+			'section'  => 'rt_scripts_optimizer_settings_section'
+		]
+	];
+
+	/**
+	 * Add fields from settings array to following core functions.
+	 * 
+	 * add_settings_field( $id, $title, $callback, $page, $section )
+	 * - https://developer.wordpress.org/reference/functions/add_settings_field/
+	 * 
+	 * register_setting( $option_group, $option_name )
+	 * - https://developer.wordpress.org/reference/functions/register_setting/
+	 * 
+	 */
+	foreach ( $script_settings as $option_name => $settings ) {
+
+		add_settings_field(
+			$settings[ 'id' ],
+			$settings[ 'title' ],
+			$settings[ 'callback' ],
+			$settings[ 'page' ],
+			$settings[ 'section' ]
+		);
+
+		register_setting( 'rt-scripts-optimizer-settings', $option_name );
```

Comment:
Hi @sudhir-yadav @pradeep910, This is the exact same code that we had earlier just shorten by adding a foreach loop. I think we need to register only 1 settings and fetch other settings by array key. Something similar to below code.

```
// Define default array
$defaults = array(
   'setting_1' => 'default_value_1',
   'setting_2' => 'default_value_2',
);

// Register the setting
register_setting( 'my_settings_group', 'my_array_setting', array(
   'type' => 'array',
   'default' => $defaults,
) );

// Update code that uses the setting
$options = get_option( 'my_array_setting', $defaults );
$setting_1 = $options['setting_1'];
$setting_2 = $options['setting_2'];
```

Of course we need to handle for the settings that are saved with standalone options after implementing above. This is my suggestion, and it would best because we need to call just one option and that would be stored in an array we can fetch by array key.

Reply:
Also settings callback would be same, there would be no change, just option fetch from settings key.